exporter/redis.go

```diff
@@ -436,16 +436,31 @@ func (e *Exporter) scrape(scrapes chan<- scrapeResult) {
 		defer c.Close()
 		log.Debugf("connected to: %s", addr)
 
+		is_cluster := false
+
 		info, err := redis.String(c.Do("INFO", "ALL"))
 		if err == nil {
 			err = e.extractInfoMetrics(info, addr, scrapes)
-		}
-		if err != nil {
+			if strings.Index(info, "cluster_enabled:1") != -1 {
```

Comment:
Why not move that `if` check to line 453 and only do the `CLUSTER INFO` if the first `extractInfoMetrics` in 443 succeeded? That way you also won't need the extra `is_cluster` variable.